### PR TITLE
Invoke block when paginate returns just one page

### PR DESCRIPTION
**Pull request by StoneFrog**

### CODE REVIEW

**Comment by Mandaryn** [comment]
@StoneFrog looks messy imho, could we separate the autopaginate handling from block handling?
if autopaginate do autopaginate stuff
if block do block stuff
makes sense?

**Comment by Mandaryn** [comment]
not yet there, the whole initial call should also be part of the recurrence
and then the non autopaginate case is just like autopaginate one just terminated at first loop

**Comment by StoneFrog** [comment]
To be honest I can't see how that would make things clearer. Why initial call should be part of the recurrence if it's always called only once no matter whether we use autopagintaion, block or single fetch? Would't that make it necessary to add another conditional inside new method and increase its complexity? Also, since initial response is required for both pagination with block and autopaginate, and seems like unnecessary duplication to use it in both methods.

**Comment by Mandaryn** [comment]
because its always recurrence then, just happens once is this specific case :P, imho the default behaviour for method paginate is to paginate the entire collection, and if autopaginate is false and no block given it just stops at the first page. Imho will just look much better with recurrence only.

File: lib/bookingsync/api/client.rb [hunk 1]
```diff
@@ -361,5 +347,38 @@ def self.instrument(name, payload = {})
...
+
+    def initial_call(path, options, request_settings)
+      request_method = request_settings[:request_method]
+      if request_method == :get
+        response = call(request_method, path, query: options)
+      else
+        response = call(request_method, path, options)
+      end
+      @pagination_first_response = response
+      response
```
Comment (LeonidMorozov) [inline lib/bookingsync/api/client.rb]: do we really need this line?

Comment (StoneFrog) [inline lib/bookingsync/api/client.rb]: you're right we don't

**Pull request merged**